src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -112,6 +123,30 @@ public void deleteIndex(String indexName) throws PineconeException {
     }
 
     public CollectionModel createCollection(CreateCollectionRequest createCollectionRequest) throws PineconeException {
+        if (createCollectionRequest == null) {
+            throw new PineconeValidationException("CreateCollectionRequest object cannot be null");
+        }
+        CollectionModel collection = null;
+        try {
+            collection = manageIndexesApi.createCollection(createCollectionRequest);
+        } catch (ApiException apiException) {
+            handleApiException(apiException);
+        }
+        return collection;
+    }
+
+    public CollectionModel createCollection(String collectionName, String sourceIndex) throws PineconeValidationException {
+        if (collectionName == null || collectionName.isEmpty()) {
+            throw new PineconeValidationException("collectionName cannot be null or empty");
+        }
+        if (sourceIndex == null || sourceIndex.isEmpty()) {
+            throw new PineconeValidationException("sourceIndex cannot be null or empty");
+        }
```

Comment:
Ah I'm seeing why exposing the native type functions along with the `CreateCollectionRequest` can be weird.

If we want to expose different arrangements of arguments for these functions, we should try and keep the validation in one spot. The way these functions are now we'd still technically need to validate for `name` and `source` within the `CreateCollectionRequest` object since they can pass that directly. So ideally all the overloads would call one function where we do the actual validation.

So something like:

```java
public CollectionModel createCollection(CreateCollectionRequest createCollectionRequest) throws PineconeException {
  if (createCollectionRequest == null) {
      throw new PineconeValidationException("CreateCollectionRequest object cannot be null");
  }
  if (createCollectionRequest.getName() == null || createCollectionRequest.getName().isEmpty()) {
      throw new PineconeValidationException("name cannot be null or empty");
  }
  if (createCollectionRequest.getSource() == null || createCollectionRequest.getSource().isEmpty()) {
      throw new PineconeValidationException("source cannot be null or empty");
  }
  CollectionModel collection = null;
  try {
      collection = manageIndexesApi.createCollection(createCollectionRequest);
  } catch (ApiException apiException) {
      handleApiException(apiException);
  }
  return collection;
}

public CollectionModel createCollection(String name, String source) {
  CreateCollectionRequest createCollectionRequest = new CreateCollectionRequest().name(name).source(source);
  
  return createCollection(createCollectionRequest);
}
```

But then we lose the specificity of the error message containing the named arguments like `collectionName` / `sourceIndex`. This is probably why we abstract away the request class in `Index` and `AsyncIndex` and use all the explicit types, validate, and then build the request object at the end. I'm not really sure what's best here, maybe @ssmith-pc has some advice on what makes sense.

Reply:
Ohhh I see. So only validate in the OG `createCollection` method?

Reply:
(Resolved IRL)